packages/node/src/retryHandler.ts

```diff
@@ -111,8 +111,39 @@ export class RetryHandler {
     }
   }
 
-  private _onEventsError(events: ReadonlyArray<Event>): void {
-    events.forEach((event: Event) => {
+  private _onEventsError(events: ReadonlyArray<Event>, response: Response): void {
+    let eventsToRetry: ReadonlyArray<Event> = events;
+    switch (response.body?.code ?? response.statusCode) {
+      case 429:
+        if (response.body?.code === 429) {
```

Comment:
This is a mistake with the specification of the types. I'll tighten the `Response` type so that the code returned from the body and the one in the response itself match.

Reply:
Oh, so you were using the body `code` to restrict what needed to be inside the union type?

I can appreciate not wanting to have a giant union type definition.

Reply:
right, there was not a typing that enforce anything between `response.statusCode` and `response.body.code`. I was actually in the process create that giant union type and removing `response.body.code` - do you think that's a good idea? it might be more code in our types, but it might be easier for other developers to use + reason about these types?

Reply:
Maintainability of the types is a real consideration. If they're too hard for us to reason about just in order for us to make them a little more correct it might be too much of a trade-off. Despite that I think that it's probably worth continuing in this case. Having a more correct type is particularly useful for our customers here as well as helping us.

Reply:
that makes a lot of sense. I was going back and forth between making the types tougher, and I think I arrived at at least tying the `Status` and `ResponseBody` together for each `success`, `invalid`, etc. and making sure my code was consistent with always trying to read the `Status` and not the `StatusCode`. Think it helped make the code a little easier to read.